src/runtime/runtime_stm32f103.go

```diff
@@ -117,14 +117,7 @@ func nanosecondsToTicks(ns int64) timeUnit {
 
 // sleepTicks should sleep for specific number of microseconds.
 func sleepTicks(d timeUnit) {
-	for d != 0 {
-		ticks()            // update timestamp
-		ticks := uint32(d) // current scaling only supports 100 usec to 6553 msec
-		if !timerSleep(ticks) {
-			return
-		}
-		d -= timeUnit(ticks)
-	}
```

Comment:
This code is necessary. It makes sure that you can sleep for longer durations than the timer supports.